docs/tutorial/voting-tutoral.md

```diff
@@ -0,0 +1,214 @@
+
+Before deploying the Voting smart dApp on our private shard, you must first generate a Calimero [auth token](/docs/getting_started/generate_token.md). This token will authenticate and authorize external applications to communicate with our shard.
+To create an auth token, follow these steps:
+
+- Log in to your Calimero Network account.
+- Click on the **Security** dropdown in the left navigation menu.
+- Select **Tokens**.
+- Click on **Create new token**.
+- Choose a name for your auth token.
+- Select a duration for the token. For this demo, we will use a Perpetual token.
+- Configure the token's permissions by selecting the appropriate checkboxes. In this case, we will select **Full access**.
+
+![](../../static/img/voting_token.png)
+
+:::tip
+Make sure to store your token in a safe place, as it will be required to communicate with the private shard and cannot be obtained afterwards. For more information, see [Generate token](/docs/getting_started/generate_token.md).
+:::
+
+## Setting up the near CLI
+
+To interact with Calimero shard using `near-cli`, you need to set the token value using **near set-api-key** command. 
+
+- Set API key
+
+```bash
+ near set-api-key https://rpc.testnet.near.org <AUTH_TOKEN>   
+```
+
+Replace **<AUTH_TOKEN>** with your token value
+
+:::note
+The https://rpc.testnet.near.org value is currently ignored by the CLI. However, this issue is fixed and will be released in the next version of the CLI.
+:::
+
+- Create a new keypair for the shard main account (if your shard name is 'demos-calimero-testnet', your main account is 'demos.calimero.testnet').
+
+```bash
+ near generate-key SHARD_ID.calimero.testnet --networkId SHARD_ID-calimero-testnet
+```
+
+Replace **<SHARD_ID>** with your shard name
+
+For our example this will be:
+
+```bash
+near generate-key demos.calimero.testnet --networkId demos-calimero-testnet
+```
+
+This command will create a keypair in the `~/.near-credentials/` folder.
+
+- Go to the `~/.near-credentials/` folder to view the  keypair file which is usually in a `.json` format.
+
+```bash
+cd ~/.near-credentials/network-id/account-id.json
+```
+
+For our example, this will be:
+
+```bash
+cd ~/.near-credentials/network-id/demos-calimero-testnet
+```
+
+From the `.json` file, you'll see the account _id, private-key and public-key
+
+## Link public key to console
+
+From the `.json` file, you'll see the account _id, private-key and public-key. You'll need to copy the public key from the .json and link it to your account in the console.
+
+- Open Calimero [Console](https://app.calimero.network/dashboard)
+- Click on **Security**
+- Click on **Custodial Accounts**
+- Search for your custodian account ID in the list and click on the  `⋮` three dots menu.
+- Click on Add public key
+
+![](../../static/img/public_keys.png)
+
+## Create sub account
+
+Next, we will create sub account which will be used to deploy the account that we have previously built. Run the following command:
```

Comment:
These steps (creating a pubKey, adding it to shard master account, then creating a subaccount) can be done from the console, i.e. you can create a subaccount from the console.
Then you need generate a new keypair and add it to the subaccount (also using the console).

Then the mentioned `deploy_calimero.sh` is entirely useable, no need to use only a part of the script.